NextcloudTalk/NewRoomTableViewController.swift

```diff
@@ -0,0 +1,303 @@
+            let contactsForIndex = contacts[index]
+            guard let contact = contactsForIndex?[indexPath.row] else {
+                return UITableViewCell()
+            }
+
+            let contactCell = tableView.dequeueReusableCell(withIdentifier: kContactCellIdentifier) as? ContactsTableViewCell ??
+            ContactsTableViewCell(style: .default, reuseIdentifier: kContactCellIdentifier)
+
+            contactCell.labelTitle.text = contact.name
+
+            let contactType = contact.source as String
+            contactCell.contactImage.setActorAvatar(forId: contact.userId, withType: contactType, withDisplayName: contact.name, withRoomToken: nil)
+
+            return contactCell
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if searchController.isActive && !resultTableViewController.contacts.isEmpty {
+            let index = resultTableViewController.indexes[indexPath.section]
+            let contactsForIndex = resultTableViewController.contacts[index]
+            guard let contact = contactsForIndex?[indexPath.row] else { return }
+
+            self.createRoomWithContact(contact)
+        } else if indexPath.section == 0 {
+            let options = getNewRoomOptions()
+            let option = options[indexPath.row]
+
+            if option == .createNewRoom {
+                self.presentCreateRoomViewController()
+            } else if option == .joinOpenRoom {
+                self.presentJoinOpenRoomsViewController()
+            }
+
+        } else {
+            let index = indexes[indexPath.section]
+            let contactsForIndex = contacts[index]
+            guard let contact = contactsForIndex?[indexPath.row] else { return }
+
+            self.createRoomWithContact(contact)
+        }
+
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+
+    // MARK: - Actions
+
+    func presentCreateRoomViewController() {
+        let viewController = RoomCreationTableViewController(account: account)
+        let navController = NCNavigationController(rootViewController: viewController)
+        self.present(navController, animated: true)
+    }
+
+    func presentJoinOpenRoomsViewController() {
+        let viewController = OpenConversationsTableViewController()
+        let navController = NCNavigationController(rootViewController: viewController)
+        self.present(navController, animated: true)
+    }
+
+    func createRoomWithContact(_ contact: NCUser) {
+        NCAPIController.sharedInstance().createRoom(forAccount: account, withInvite: contact.userId, ofType: .oneToOne, andName: nil) { room, error in
+            if let token = room?.token, error == nil {
+                self.navigationController?.dismiss(animated: true, completion: {
+                    NotificationCenter.default.post(name: NSNotification.Name.NCSelectedUserForChat, object: self, userInfo: ["token": token])
+                })
```

Comment:
Could also write it like: 

```suggestion
                self.navigationController?.dismiss(animated: true) {
                    NotificationCenter.default.post(name: NSNotification.Name.NCSelectedUserForChat, object: self, userInfo: ["token": token])
                }
```